packages/api-client/src/v2/hooks/use-workspace-client-events.ts

```diff
@@ -75,6 +80,9 @@ export const useWorkspaceClientEvents = ({
   /** Use router for some redirects */
   const router = useRouter()
 
+  /** Use route for the path variables */
+  const route = useRoute()
```

Comment:
Lets decouple the state from this hook. As discussed in the call we will fast follow:

- move hook to stateless in the shared hooks folder
- add onBefore, onAfter events to the hook
- Separate wrapper hooks in `entry-points/app` and `entry-points/modal`